### PR TITLE
Divide by Zero Exception fixed; Added Log2(x) feature

### PR DESCRIPTION
**Pull request by SRvSaha**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -2,122 +2,181 @@
...
 import math
+import sys
 
 "Performs addition"
-def add(a,b):
-    val = a+b
+
+
+def add(a, b):
+    val = a + b
```
Comment (trishnaguha) [inline main.py:11]: Can you move the docstring inside method? and same for rest of the methods?

**Review by trishnaguha — CHANGES_REQUESTED** [review]

File: main.py [hunk 2]
```diff
@@ -2,122 +2,181 @@
...
     return math.radians(a)
 
 "The Main function"
+
+
 def main():
-    parser = argparse.ArgumentParser(description = "A Command Line Calculator", epilog = "Enter two numbers for addition,subtraction and so on....\n Enter only one number for percentage calculation,factorial and so on....")
-	
+    parser = argparse.ArgumentParser(description="A Command Line Calculator",
+                                     epilog="Enter two numbers for addition,subtraction and so on....\n Enter only one number for percentage calculation,factorial and so on....")
```
Comment (trishnaguha) [inline main.py]: It would be nice if we can follow PEP8 rule. Could you break the text in couple of lines?

**Review by trishnaguha — CHANGES_REQUESTED** [review]

File: main.py (hunk 1 above)
Comment (SRvSaha) [reply main.py:11]: Well, I can do that. Will be updating the asked changes in some time.

**Comment by trishnaguha** [comment]
Otherwise Looks good to me.

File: main.py (hunk 2 above)
Comment (SRvSaha) [reply main.py]: Actually, I did Auto PEP8 formatting using Anaconda plugin in Sublime Text. 

Anyways, I will do the changes. :) Thanks for giving me an opportunity

**Comment by SRvSaha** [comment]
image

I think there's something wrong going on behind the scene.
As per the code there should have been a new line after so on... And the Enter only one num bla bla should come in new line. But why is this "\n" not working ? Should I keep it the way it is any fix the issue of "line too long " according to PEP8 or should I work on the "\n" thing as well??
<code>parser = argparse.ArgumentParser(description="A Command Line Calculator",
                                     epilog="Enter two numbers for addition,subtraction and so on....\n Enter only one number for percentage calculation,factorial and so on....")
</code>
"

**Comment by SRvSaha** [comment]
Glad to fix all the issues pointed out by you and the ones I found that needed to be fixed.

Fixed Issues :
1. PEP8 Guideline-based DOCSTRING ERROR and LINE TOO LONG ERROR.
2. Added formatter class for epilog in argument parser for better readability & fixed the newline issue.
3. The percentage method is modified a bit so that the division is more generic(support of Python2 in Testing).

Hoping for a merge for my first contribution to Open Source :)

**Comment by SRvSaha** [comment]
Changed made :
1. Added feature of log2(x) which is very important since most of the times we deal with log2(x) in Computer Science.
2. Updated Readme file with the logarithm2(x)
3. Fixed all PEP8 errors (except docstring missing in test_main.py) in the codebase.

**Comment by trishnaguha** [comment]
Looks good! Thanks @SRvSaha .

**Pull request merged**